syntax = "proto3";

// TODO:
// Come up with a better name for this.
package beavertail;

message DatagramPush {
  string busID                    = 10;
  uint64 passengerCount           = 20;
  double passengerCountConfidence = 30;
  double latitude                 = 40;
  double longitude                = 50;
  int64  timestamp                = 60;  // Unix timestamp
}

message DatagramAck {
  enum AckType {
    OK   = 0;
    BAD  = 1;
    BUSY = 2;
  }
  AckType acknowledgment = 10;
}

// gRPC defines services in this case PushDatagram.
service PushDatagram {
  // In this service we expose an RPC called Push
  // it takes in an DatagramPush and returns an DatagramAck
  // dont ask me about the `{}` bc idk :(
  rpc Push(DatagramPush) returns (DatagramAck) {}
}
